### PR TITLE
Dex: Token Input

### PR DESCRIPTION
**Pull request by 0xalecks**
Token Input component so we can work in parallel

### CODE REVIEW

File: projects/dex-ui/src/components/Swap/TokenInput.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
+import React, { MouseEvent, useCallback, useRef, useState } from "react";
+import { Token, TokenValue } from "@beanstalk/sdk";
+import { FC } from "src/types";
+import styled from "styled-components";
+import { BasicInput } from "./BasicInput";
+import { TokenPicker } from "./TokenPicker";
+import { reactWhatChanged as RWC } from "src/utils/what-changed";
```
Comment (marshall-bean) [inline projects/dex-ui/src/components/Swap/TokenInput.tsx:7]: Is this even being used? If not can we remove the whole thing?

File: projects/dex-ui/src/utils/what-changed.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+/** is pojo or array */
+function isObj(obj: any) {
+  return typeof obj === "object" && obj !== null;
+}
+
+/**
+ * compares a react dependencies list with its previous values (from previous cycle)
+ * and prints which dependency is the one that changed
+ */
+export const reactWhatChanged = function reactWhatChanged(dependencies: AnyObject, verbose: boolean = false, id?: string | number) {
```
Comment (marshall-bean) [inline projects/dex-ui/src/utils/what-changed.ts:17]: 🤔  What's all this about? Looks like we aren't using this?

**Review by marshall-bean — CHANGES_REQUESTED** [review]

File: projects/dex-ui/src/index.tsx [hunk 3]
```diff
@@ -3,12 +3,13 @@ import ReactDOM from "react-dom/client";
 import { App } from "src/components/App/App";
 import { Wrapper } from "src/components/App/Wrapper";
 
+import "src/normalize.css";
 import "src/index.css";
 
 ReactDOM.createRoot(document.getElementById("root")!).render(
-  <React.StrictMode>
+  // <React.StrictMode>
```
[the remark this answers is not in the document]
Comment (0xalecks) [reply projects/dex-ui/src/index.tsx:10]: no it's good for dev.. i'll re-enable it in next pr

File: projects/dex-ui/src/components/Frame/Frame.tsx [hunk 4]
```diff
@@ -46,33 +61,38 @@ const Container = styled.div`
   flex-direction: column;
   box-sizing: border-box;
   width: 100vw;
+  align-items: center;
 `;
 
-const NavContainer = styled.div`
-  // border: 1px solid blue;
+const NavContainer = styled.nav`
+  // border-bottom: 1px solid gray;
```
[the remark this answers is not in the document]
Comment (0xalecks) [reply projects/dex-ui/src/components/Frame/Frame.tsx:68]: doesn't matter right now, whole nav will get a restyle once design is ready

File: projects/dex-ui/src/utils/what-changed.ts (hunk 2 above)
Comment (0xalecks) [reply projects/dex-ui/src/utils/what-changed.ts:17]: This is a useful function to spit out what changed in a dependency array. Was trying to figure out what was causing extra renders, and this showed me.

```
useEffect(() => {
  // use one, two
}, reactWhatChanged([one, two]))
```

File: projects/dex-ui/src/components/Swap/TokenInput.tsx (hunk 1 above)
Comment (0xalecks) [reply projects/dex-ui/src/components/Swap/TokenInput.tsx:7]: not anymore, will remove this import

**Pull request merged**